ScrapeRepos.ts

```diff
@@ -0,0 +1,356 @@
+const axios = require('axios');
+const cheerio = require('cheerio');
+const time = require('console');
+const showdown = require('showdown');
+
+
+/** Fetches the repo owner and name for each repo on the GitHub trending page
+ * @param {number} timeMode 0: daily; 1: weekly; 2: monthly
+ * @returns {string[]} an array that stores alternatingly the owner and the name of each repo
+ */
+async function fetchRepos( timeMode: number) {
+  try {
+    const timeModes = ['daily', 'weekly', 'monthly'];
+    const response = await axios.get('https://github.com/trending?since=' + timeModes[timeMode]);
+    const $ = cheerio.load(response.data);
+    const repos: string[] = [];
+
+    $('h2 a').each((i: number, el: cheerio.Element) => {
+      const repoName = $(el).text().trim();
+      repos.push(repoName);
+    });
+    
+    var trendingSplit: string[] = [];
+    // trim the repos to be correctly formatted
+    repos.forEach((repo) => {
+      let trimmedName = repo.replace(/\n\s+/g, '');
+      trimmedName = trimmedName.replace(/\//g, "");
+      const stringSplit = trimmedName.split(" ");
+      trendingSplit.push(stringSplit[0]);
+      trendingSplit.push(stringSplit[1]);
+    });
+
+    return trendingSplit;
+  } catch (error) {
+    console.error(error);
```

Comment:
For the security purpose, logs should not be printed.

Reply:
Should we already implement this? It will be harder to actually find errors without printing them where they origniated from. 
But if you think it's a good idea I also understand why and I will implement it

Reply:
Should we already implement this? The errors could be much harder to understand when not printed where they originated from.
But if you are sure that this is a good idea I understand why and I can implement this throughout the code.